sdk/identity/identity/test/internal/node/managedIdentityCredential.spec.ts

```diff
@@ -232,12 +226,16 @@ describe("ManagedIdentityCredential", function() {
       ...mockHttpClient.tokenCredentialOptions
     });
 
-    const promise = credential.getToken("scopes");
+    // Sinon's clock has some issues with Node 16.
+    // If we remove this conditional on Node 16, we get the following error:
+    //   PromiseRejection HandledWarning: Promise rejection was handled asynchronously
+    // It will point to the `await promise` line below.
+    const clock = process.version.startsWith("v16") ? undefined : sandbox.useFakeTimers();
```

Comment:
Could be a nice follow up item to submit a bug to sinon.

Reply:
What is the point of using the clock?

Reply:
I am trying to narrow this down to an actual bug outside of our tests. It’s a bit challenging. Once I have it narrowed down, I’ll make an issue on Sinon (or perhaps do changes in our code to fix it? I don’t know yet).